### PR TITLE
Update creating_gem.en.html.md

### PR DESCRIPTION
**Pull request by Mth0158**
Making a proposal to update the wiki from this issue: [Issue 4458](https://github.com/rubygems/rubygems/issues/4458)

The descriptions of the fields to complete come from Rubygems.

Mth0158

### What was the end-user problem that led to this PR?

If metadata["allowed_push_host"] is not commented and you are not using your own ruby gem server you will not be able to push your gem to rubygems.org. 
This causes an error that is not displayed on the CLI and is very confusing.

### What was your diagnosis of the problem?

The diagnosis was previously made on the issue.

### What is your fix for the problem, implemented in this PR?

To explicitly write in the wiki to comment this line in gemspec file if you do not use your own gem server.

### Why did you choose this fix out of the possible options?

Making the tutorial a bit more user-friendy.

### CODE REVIEW

**Comment by Mth0158** [comment]
It also might be a plus to change the line in the gemspec for people that skip the tutorial.

`spec.metadata["allowed_push_host"] = "TODO: Set to 'http://mygemserver.com'"`

=> `spec.metadata["allowed_push_host"] = "TODO: Set to 'http://mygemserver.com' or comment this line if your are not using a gem server to push your gem"`

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 1]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
+      * *homepage:* The URL of your gem's home page, it can be the URL of its website or public repo
+      * *metadata["allowed_push_host"]:* The field allows you to restrict gem pushes to a single host. If you are not running your own gem server you should comment this line, otherwise you will not be able to push your gem to rubygems.org.
+      * *metadata["source_code_uri"]:* The URL of your gem's public repo
+      * *metadata["changelog_uri"]:* The URL of your gem's CHANGELOG.md
```
Comment (olleolleolle) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:92]:
```suggestion
      * `metadata["changelog_uri"]`: The URL of your gem's CHANGELOG.md.
```

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 2]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
+      * *homepage:* The URL of your gem's home page, it can be the URL of its website or public repo
+      * *metadata["allowed_push_host"]:* The field allows you to restrict gem pushes to a single host. If you are not running your own gem server you should comment this line, otherwise you will not be able to push your gem to rubygems.org.
+      * *metadata["source_code_uri"]:* The URL of your gem's public repo
```
Comment (olleolleolle) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:91]:
```suggestion
      * `metadata["source_code_uri"]`: The URL of your gem's public repo.
```

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 3]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
+      * *homepage:* The URL of your gem's home page, it can be the URL of its website or public repo
```
Comment (olleolleolle) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:89]:
```suggestion
      * `homepage`: The URL of your gem's home page, it can be the URL of its website or public repo.
```

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 4]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
```
Comment (olleolleolle) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:88]:
```suggestion
      * `description` (optional): A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
```

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 5]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
```
Comment (olleolleolle) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:87]:
```suggestion
      * `summary`: A short summary of your gem's description.
```

File: source/localizable/v2.2/guides/creating_gem.en.html.md [hunk 6]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
...
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
+      * *homepage:* The URL of your gem's home page, it can be the URL of its website or public repo
+      * *metadata["allowed_push_host"]:* The field allows you to restrict gem pushes to a single host. If you are not running your own gem server you should comment this line, otherwise you will not be able to push your gem to rubygems.org.
```
Comment (simi) [inline source/localizable/v2.2/guides/creating_gem.en.html.md:90]: > running your own gem server

That's really advanced stuff for beginner. Is there any chance to reword this to make it clear by allowing `rubygems.org` you'll be able to push to public (accessible by everyone) repository and this is needed for this guide to be able to publish in the end?

**Comment by deivid-rodriguez** [comment]
Should we comment out this line, or even remove it? It seems like a small feature only useful for private gems that makes things more inconvenient for the rest of the world.

**Comment by indirect** [comment]
For what it's worth, we added this line because there was a high support load of helping people remove private gems that they didn't mean to push to rubygems.org.

It's very confusing that the error is silent, but I don't know if just letting people accidentally publish code they need our help to remove is better. 😅

**Comment by deivid-rodriguez** [comment]
Right, I missed that context. Suggestion retired then :)

I think changing the line from `http://mygemserver.com` to `https://mygemserver.com` might do the trick, then?

**Comment by indirect** [comment]
Hmmm.... maybe we could make the private config line an option? "DO YOU WANT ALL GEMS YOU MAKE TO BE PUBLIC Y/N IF YOU DO THIS WE WILL NOT HELP YOU WHEN YOU ACCIDENTALLY PUBLISH YOUR COMPANY'S PRIVATE CODE" 😅

**Comment by deivid-rodriguez** [comment]
@indirect A setting to configure whether to generate the `allowed_push_host` setting or not sounds good to me! But using https in the current placeholder would also alleviate the issue in any case, and is straightforward, so I would start with that.

**Comment by indirect** [comment]
Great, let’s do it. 🙂

**Comment by deivid-rodriguez** [comment]
@sonalkr132 [beat me to it](https://github.com/rubygems/rubygems/pull/4671) :smiley:

**Comment by tnir** [comment]
Hi @Mth0158. In #625, I moved the target file `source/localizable/v2.2/guides/creating_gem.en.html.md` to `source/localizable/guides/creating_gem.en.html.md`, so could you mind updating your change? You can do it just move the file to `source/localizable/guides`.

**Comment by Mth0158** [comment]
Done @tnir ✌🏼

File: source/localizable/guides/creating_gem.en.html.md [hunk 7]
```diff
@@ -83,6 +83,13 @@ extension and the _.bundle_ directory.
  * **foodie.gemspec**: The Gem Specification file. This is
 where we provide information for Rubygems' consumption such as the name, description and homepage 
 of our gem. This is also where we specify the dependencies our gem needs to run.
+    * Fields to complete:
+      * *summary:* A short summary of your gem's description.
+      * *description (optional):* A long description of your gem. The description should be more detailed than the summary but not excessively long. A few paragraphs is a recommended length with no examples or formatting.
+      * *homepage:* The URL of your gem's home page, it can be the URL of its website or public repo
+      * *metadata["allowed_push_host"]:* The field allows you to restrict gem pushes to a single host. If you are not running your own gem server you should comment this line, otherwise you will not be able to push your gem to rubygems.org.
+      * *metadata["source_code_uri"]:* The URL of your gem's public repo
+      * *metadata["changelog_uri"]:* The URL of your gem's CHANGELOG.md
```
Comment (olleolleolle) [inline source/localizable/guides/creating_gem.en.html.md:92]: Cool addition.

Here's a small tweak:

```suggestion
      * *metadata["source_code_uri"]:* The URL of your gem's public repo.
      * *metadata["changelog_uri"]:* The URL of your gem's CHANGELOG.
```

- Ending sentences in a period.
- Avoiding deciding on what file format a changelog has.

**Comment by tnir** [comment]
@Mth0158 Could you address file conflicts to get this ready? 🙏

**Comment by tnir** [comment]
@Mth0158 Could you address file conflicts to get this ready? 🙏
This is a friendly reminder.

**Comment by Mth0158** [comment]
Hi @tnir,

I have rebased the branch onto master, but pushed a new [PR](https://github.com/rubygems/bundler-site/pull/1067). I am closing this one so we can finish the work on the newly created one.

Let me know if further changes are needed, or if you need any help on other subjects,
